attempt list react/desktop

Mandvii opened the pull request:
Attempt List Task
1-Person List
2-Sesssion List
3-Statement List

Mandvii committed:
Task-Attempt List
Add attempts tab to content detail screen

Mandvii committed:
attempts tab added and attempt person list showing

Mandvii committed:
attempts tab added and attempt person list showing

Mandvii committed:
attempts tab added and attempt person list showing

Mandvii committed:
Update

Mandvii committed:
Update

Mandvii committed:
Update

Prashant-Android committed:
added social warning in MessageListScreen and ClazzAssignmentDetailOverviewScreen .

Prashant-Android committed:
added social warning strings in strings.xml

Prashant-Android committed:
added social warning strings in strings.xml

Mandvii committed:
SessionList

Prashant-Android committed:
removed hardcoded color value from SocialWarningListItem.kt

Prashant-Android committed:
removed hardcoded color value from SocialWarningListItem.kt

Mandvii committed:
Session List

Mandvii committed:
Session List

Mandvii committed:
Session List

Mandvii committed:
Session List

Mandvii committed:
Session List

Mandvii committed:
Session List

Mandvii committed:
Session List

Mandvii committed:
Session List fix

Mandvii committed:
Session List fix

Prashant-Android committed:
correct naming convention

Prashant-Android committed:
correct naming convention

Prashant-Android committed:
correct naming convention

Prashant-Android committed:
correct naming convention

Prashant-Android committed:
Fixed binding issue in jvmTest

Mandvii committed:
Statement List

Mandvii committed:
Statement List

Prashant-Android committed:
Adjust Ui.

Mandvii committed:
Attempts List Task

Mandvii committed:
Attempts List Task

Mandvii committed:
Attempts List Task

Mandvii committed:
Attempts List Task

Mandvii committed:
Attempts Person List Task

Mandvii committed:
Attempts Person List Task

Mandvii committed:
Attempts Person List Task

Mandvii committed:
Attempts Person List Task

Mandvii committed:
Attempts List Task

Mandvii committed:
Attempts List Task

Mandvii committed:
Attempts List Task

Mandvii committed:
Attempts List Task

Prashant-Android committed:
Reverted unintended formatting changes in unrelated files.
Removed unnecessary Paper and obsolete cautions list as per review feedback.
Updated SocialWarningListItem layout to match task requirements with headline, secondary text, and buttons below text.
Removed unnecessary Paper and obsolete cautions list as per review feedback.

Prashant-Android committed:
Reverted unintended formatting changes in unrelated files.
Removed unnecessary Paper and obsolete cautions list as per review feedback.
Updated SocialWarningListItem layout to match task requirements with headline, secondary text, and buttons below text.
Removed unnecessary Paper and obsolete cautions list as per review feedback.

Prashant-Android committed:
Fix String issue & social warning url.

Prashant-Android committed:
Fix String issue & social warning url.

Prashant-Android committed:
Fix String issue & social warning url.

Mandvii committed:
Attempt List HttpAccessible

Mandvii committed:
Attempt List HttpAccessible

Mandvii committed:
Attempt list

Mandvii committed:
Attempt list

Mandvii committed:
Attempt list

Mandvii committed:
Attempt list

Mandvii committed:
Merge branch 'primary' into dev-attempt-list-react

Prashant-Android committed:
- Moved `SOCIAL_WARNING_WEB_URL` to `ShowSocialWarningUseCase`
- Buttons are part of secondary text with proper alignment in a responsive layout.

Prashant-Android committed:
- Moved `SOCIAL_WARNING_WEB_URL` to `ShowSocialWarningUseCase`
- Buttons are part of secondary text with proper alignment in a responsive layout.

Mandvii committed:
Attempt list

Mandvii committed:
Attempt list

Mandvii committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

Prashant-Android committed:
Fixed ui.

Prashant-Android committed:
Fixed ui.

mikedawson committed:
Update CODING-STYLE.md

Prashant-Android committed:
Fixed ui.

mikedawson committed:
Update Android Gradle Plugin version.

mikedawson committed:
Fix ClazzAssignmentDetailOverviewScreen on app-react: React hook from Social warning was used within an if statement.

Mandvii committed:
Desktop/Android-

Attempts Person List Screen:
(Person name, person image,
completion, score)

Mandvii committed:
Desktop/Android-

Attempts Person List Screen:
(Person name, person image,
completion, score)

Mandvii committed:
Desktop/Android-

Attempts Session List Screen:
(Pass fail status, score, completion percentage)

Mandvii committed:
Desktop/Android-

Attempts Session List Screen:
(Pass fail status, score, completion percentage)

Mandvii committed:
Desktop/Android-

Attempts Session List Screen:
(Pass fail status, score, completion percentage)

Mandvii committed:
Desktop/Android-

Attempts Session List Screen:
(Pass fail status, score, completion percentage)

Mandvii committed:
react-

Attempts Person Session List Screens

Mandvii committed:
react-

Attempts Person Session List Screens

Mandvii committed:
desktop-
Attempts person list screen: showing completion bar or score bar accordingly

Mandvii committed:
desktop-
Attempts person list screen: showing completion bar or score bar accordingly

Mandvii committed:
react-
Attempts person list screen: showing completion bar or score bar accordingly

Mandvii committed:
react-
Attempts person list screen: showing completion bar or score bar accordingly

Mandvii committed:
Desktop-
Attempts session list screen: showing pass,fail,complet,incomplete ,completion percentage and score  accordingly

Mandvii committed:
Desktop-
Attempts session list screen: showing pass,fail,complet,incomplete ,completion percentage and score  accordingly

Mandvii committed:
react-
Attempts session list screen: showing pass,fail,complet,incomplete ,completion percentage and score  accordingly

Mandvii committed:
react-
Attempts session list screen: showing pass,fail,complet,incomplete ,completion percentage and score  accordingly

Mandvii committed:
react-
Attempts session list screen: showing pass,fail,complet,incomplete ,completion percentage and score  accordingly

Mandvii committed:
react-
Attempts session list screen: showing pass,fail,complet,incomplete ,completion percentage and score  accordingly

Mandvii committed:
get Statement Entity

Mandvii committed:
get Statement Entity

Mandvii committed:
get Statement Entity
Joining StatementEntity and verb entity

Mandvii committed:
get Statement Entity
Joining StatementEntity and verb entity

mikedawson committed:
Add queries for list of persons who attempted a given piece of content nad list of sessions by person and content uid.

mikedawson committed:
Add queries for list of persons who attempted a given piece of content nad list of sessions by person and content uid.

mikedawson committed:
Add query for list of statements by registration uid.
Update SessionTimeAndProgressInfo to include registration uid to be passed to next screen.

mikedawson committed:
Add query for list of statements by registration uid.
Update SessionTimeAndProgressInfo to include registration uid to be passed to next screen.

mikedawson committed:
Add isSuccessful and isCompleted booleans to queries for attempts list screens for list of persons and list of sessions for person queries.

mikedawson committed:
Add isSuccessful and isCompleted booleans to queries for attempts list screens for list of persons and list of sessions for person queries.

mikedawson committed:
Add max score and max progress for StatementDao#findPersonsWithAttempts

mikedawson committed:
Add max score and max progress for StatementDao#findPersonsWithAttempts

poojaustad committed:
Test description and maestro/cypress tests for attempt list test

poojaustad committed:
Test description and maestro/cypress tests for attempt list test

poojaustad committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

Mandvii committed:
Attempt List

Mandvii committed:
Attempt List

Mandvii committed:
Attempt List- changes made

Mandvii committed:
Attempt List- changes made

Mandvii committed:
Attempts Person List- desktop/android

Mandvii committed:
Attempts Person List- desktop/android

Mandvii committed:
Attempts Person List- desktop/android

Mandvii committed:
Attempts Person List- desktop/android

Mandvii committed:
Attempts Person List- react

Mandvii committed:
Attempts Person List- react

Mandvii committed:
Attempts Session List- desktop/android

Mandvii committed:
Attempts Session List- desktop/android

Mandvii committed:
Attempts Session List- desktop/android

Mandvii committed:
Attempts Session List- desktop/android

Mandvii committed:
Attempts Session List- react

Mandvii committed:
Attempts Session List- react

Mandvii committed:
Attempts Statement List- desktop/android

Mandvii committed:
Attempts Statement List- desktop/android

Mandvii committed:
Attempts Statement Lis

Mandvii committed:
Attempts Statement Lis

Mandvii committed:
findStatusForStudentsInClazz
Jenkins error:All annotations from expect `suspend fun findStatusForStudentsInClazz(clazzUid: Long, studentPersonUids: List<Long>, accountPersonUid: Long): List<BlockStatus> defined in com.ustadmobile.core.db.dao.xapi.StatementDao` must be present with the same arguments on actual `suspend fun findStatusForStudentsInClazz(clazzUid: Long, studentPersonUids: List<Long>, accountPersonUid: Long): List<BlockStatus> defined in com.ustadmobile.core.db.dao.xapi.StatementDao`, otherwise they might behave incorrectly.

Mandvii committed:
findStatusForStudentsInClazz
Jenkins error:All annotations from expect `suspend fun findStatusForStudentsInClazz(clazzUid: Long, studentPersonUids: List<Long>, accountPersonUid: Long): List<BlockStatus> defined in com.ustadmobile.core.db.dao.xapi.StatementDao` must be present with the same arguments on actual `suspend fun findStatusForStudentsInClazz(clazzUid: Long, studentPersonUids: List<Long>, accountPersonUid: Long): List<BlockStatus> defined in com.ustadmobile.core.db.dao.xapi.StatementDao`, otherwise they might behave incorrectly.

Mandvii committed:
findStatusForStudentsInClazz

Mandvii committed:
findStatusForStudentsInClazz

mikedawson committed:
Remove obsolete version migration tests from TestRoomMigration

Mandvii committed:
Successful build

Mandvii committed:
Successful build

Mandvii committed:
Duration- Desktop/Android

Mandvii committed:
Duration- Desktop/Android

Mandvii committed:
React duration

Mandvii committed:
React duration

Mandvii committed:
Time and date Session List React

Mandvii committed:
Time and date Session List React

Mandvii committed:
Time and date Session List React

Mandvii committed:
Time and date Session List React

Mandvii committed:
If no score or completion percentage in the statement list

Mandvii committed:
If no score or completion percentage in the statement list

Mandvii committed:
If no score or completion percentage in the statement list

Mandvii committed:
If no score or completion percentage in the statement list

Mandvii committed:
title updated

Mandvii committed:
title updated

poojaustad committed:
attempt list test update

poojaustad committed:
attempt list test update

poojaustad committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

mikedawson committed:
Update task_card.md

poojaustad committed:
attempt list test update

poojaustad committed:
attempt list test update

Mandvii committed:
duration-timestamp updated

Mandvii committed:
duration-timestamp updated

poojaustad committed:
attempt list test description update

poojaustad committed:
attempt list test description update

poojaustad committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

poojaustad committed:
Make sure apk file is found and installed before starting maestro test

poojaustad committed:
Make sure apk file is found and installed before starting maestro test

poojaustad committed:
Readme update

mikedawson committed:
Update README.md

mikedawson committed:
Update README.md

Clarify SDK emulator

mikedawson committed:
Update README.md

Avoid duplicate instructions on building Android apk in maestro README.

mikedawson committed:
Update README.md

Update Android Maestro README to remove duplicated instructions on running a single test.

mikedawson committed:
Update README.md

Mandvii committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempts-list

poojaustad committed:
attempt list maestro test

poojaustad committed:
attempt list maestro test

mikedawson commented:
@Mandvii this pull request has changes that are not related (e.g. the task card template and social warning). Please revert all unrelated changes (see list of files changed) and update this pull request to ensure that there are no unrelated changes

Mandvii committed:
Merge branch 'dev-attempts-list' into dev-attempt-list-react

poojaustad committed:
attempt list maestro test

poojaustad committed:
attempt list maestro test

poojaustad committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

poojaustad committed:
attempt list maestro test

poojaustad committed:
attempt list maestro test

Mandvii committed:
Changes in statement

Mandvii committed:
Changes in statement

poojaustad committed:
attempt list cypress test

poojaustad committed:
attempt list cypress test

Mandvii committed:
Changes in statement desktop

Mandvii committed:
Changes in statement desktop

Mandvii committed:
Changes in statement react

Mandvii committed:
Changes in statement react

Mandvii committed:
SessionListScreen react/desktop , UstadScreen

Mandvii committed:
SessionListScreen react/desktop , UstadScreen

Mandvii committed:
SessionListScreen react/desktop

Mandvii committed:
SessionListScreen react/desktop

mikedawson commented:
@Mandvii the pull request _still_ contains unrelated changes after the above commits:  
![Screenshot from 2024-12-18 19-34-31](https://github.com/user-attachments/assets/e2c82906-6642-4851-bf49-20c838d35c29) . This is the first priority to resolve. Please self-check using Changed Files list here on the GitHub pull request.

Mandvii committed:
Attempt Peron List

Mandvii committed:
Attempt Peron List

Mandvii committed:
Nav-Attempt Person List

Mandvii committed:
Nav-Attempt Person List

Mandvii committed:
unused methods removed from XapiSessionEntityDao

Mandvii committed:
unused methods removed from XapiSessionEntityDao

Mandvii committed:
Person

Mandvii committed:
Person

Mandvii committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

Mandvii committed:
Icon unused

Mandvii committed:
Icon unused

Mandvii committed:
Files not related to task removed

Mandvii committed:
Files not related to task removed

mikedawson commented on gradle/libs.versions.toml:
```diff
@@ -126,6 +126,7 @@ lz4 = "1.8.0"
 kmp-uuid = "0.8.4"
 zoomable = "1.6.1"
 jetbrains-kotlin-jvm = "1.9.23"
+firebase-perf-ktx = "21.0.3"
```
Why has firebase been added? This should not be here.

Mandvii replied:
removed the added firebase.

Mandvii committed:
Delete et --hard HEAD@{1}

Mandvii committed:
Delete et --hard HEAD@{1}

Mandvii committed:
Readme changes reverted

Mandvii committed:
Readme changes reverted

Mandvii committed:
Readme - primary code comparison

Mandvii committed:
Readme - primary code comparison

mikedawson committed:
Add useFormattedDuration function for Jetpack compose and update React hook.

mikedawson committed:
Add useFormattedDuration function for Jetpack compose and update React hook.

mikedawson committed:
Merge branch 'primary' into dev-attempt-list-react

mikedawson committed:
Update StatementDao query for findPersonsWithAttempts to include permission checks.

mikedawson committed:
Update StatementDao query for findPersonsWithAttempts to include permission checks.

Mandvii committed:
Duration code update-
react/desktop

If no information of resultDuration present handled

Mandvii committed:
Duration code update-
react/desktop

If no information of resultDuration present handled

Mandvii committed:
Duration code update-
react/desktop

If resultDuration is not null but formatted duration is 0hrs and 0mins ,
So have added secs for that.

Mandvii committed:
Duration code update-
react/desktop

If resultDuration is not null but formatted duration is 0hrs and 0mins ,
So have added secs for that.

Mandvii committed:
Empty State of List handled for AttemptsStatementListScreen

Mandvii committed:
Empty State of List handled for AttemptsStatementListScreen

Mandvii committed:
Empty State of List handled also if no information available visibilty handled for AttemptsSessionListListScreen

Mandvii committed:
Empty State of List handled also if no information available visibilty handled for AttemptsSessionListListScreen

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/contententry/detailattempttab/ContentEntryDeatilAttemptsSessionListScreen.kt:
```diff
@@ -0,0 +1,169 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Icon
```
All material imports must use material3 e.g. material3.Text material3.Icon , etc. Otherwise theme/appearance will be inconsistent.

mikedawson replied:
Note there is no material3.icons (vectors) - this applies to Components - e.g. Text, ListItem, Icon, etc.

Mandvii replied:
Used import of material3 for Text,Icon,ListIem,etc

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/contententry/detailattempttab/ContentEntryDeatilAttemptsSessionListScreen.kt:
```diff
@@ -0,0 +1,169 @@
+                                    }
+
+                                    else -> {
+                                        "${(attemptsSessionListItems?.maxProgress)}% Completion"
```
The app is used in multiple languages. All hardcoded strings must be replaced with using localized string resources.

Mandvii replied:
removed hardcoded strings

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/contententry/detailattempttab/ContentEntryDeatilAttemptsSessionListScreen.kt:
```diff
@@ -0,0 +1,169 @@
+                        ) {
+                            Icon(
+                                imageVector = Icons.Filled.Star,
+                                contentDescription = "Icon",
```
If contentDescription is anything other than blank or null, it must use localized strings the same as any other text used in the interface.

When an icon is used for decoration only - the contentDescription should be null or blank (so that a screen reader for someone with visual impairments will skip it). If the icon has a function (e.g. a button which has only an icon, no text, like a delete button), then the contentDescription should be set.

Mandvii replied:
Made the changes accordingly

mikedawson reviewed (changes requested):
Please make sure that all strings are localized. 
Please make sure that all compose imports for Components (Text, Icon, ListItem, etc) use material3

mikedawson committed:
Update StatementDao to fetch Statements required for findPersonsWithAttempts over http.

mikedawson committed:
Update StatementDao to fetch Statements required for findPersonsWithAttempts over http.

Mandvii committed:
Unused Files removed
Attempts Person list -
if no data available handled
if list empty handled

Mandvii committed:
Unused Files removed
Attempts Person list -
if no data available handled
if list empty handled

Mandvii committed:
Changes-
hardcode string removed , changes in material3 import and contentDescription

Mandvii committed:
Changes-
hardcode string removed , changes in material3 import and contentDescription

Mandvii committed:
StatementDao updated code

Mandvii committed:
StatementDao updated code

Mandvii committed:
removed unused firebase

Mandvii committed:
removed unused firebase

Mandvii committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

# Conflicts:
#	core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/contententry/detailattemptlisttab/ContentEntryDetailAttemptsPersonListViewModel.kt
#	lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/xapi/StatementDao.kt

poojaustad committed:
cypress attempt screen validation

poojaustad committed:
cypress attempt screen validation

Mandvii committed:
String changes and ui changes

Mandvii committed:
String changes and ui changes

Mandvii committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

sivedcodes commented on app-react/src/jsMain/kotlin/com/ustadmobile/view/contententry/detailattemptstab/ContentEntryDetailAttemptsPersonListScreen.kt:
```diff
@@ -0,0 +1,185 @@
+                                            LinearProgress {
+                                                sx {
+                                                    width = 500.px
+                                                    height = 4.px
```
Suggestion: The use of hardcoded values like 500.px and 8,4,1px and other hard coded value in the code. While these might be functional, they can make the code harder to maintain or adapt if these values need to change in the future.
According to our coding style guidelines, we should avoid hardcoding literals and instead use constants,.

Mandvii committed:
Setup download/upload task

Mandvii committed:
Setup download/upload task

Mandvii committed:
set code

Mandvii committed:
merged/rebase with primary

Mandvii committed:
merged/rebase with primary

Mandvii committed:
Merge branch 'primary' into dev-attempt-list-react

poojaustad committed:
Merge branch 'primary' into dev-attempt-list-react

poojaustad committed:
updates

Mandvii committed:
Merge branch 'refs/heads/primary' into dev-attempt-list-react

# Conflicts:
#	test-end-to-end/android-maestro/README.md

Mandvii committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

# Conflicts:
#	test-end-to-end/android-maestro/README.md

Mandvii committed:
Merge branch 'primary' into dev-attempt-list-react

# Conflicts:
#	test-end-to-end/webapp-cypress/cypress/e2e/WEB_003_005_admin_or_teacher_and_content_via_file.cy.js

Mandvii committed:
Merge branch 'primary' into dev-attempt-list-react

Mandvii committed:
moved attemptlist test file from test disable to test

poojaustad committed:
attemptlist maestro e2e test

poojaustad committed:
Merge remote-tracking branch 'origin/dev-attempt-list-react' into dev-attempt-list-react

# Conflicts:
#	test-end-to-end/android-maestro/e2e-tests/001_010_content_attempt_list_test.yaml

poojaustad committed:
Merge branch 'primary' into dev-attempt-list-react

mikedawson committed:
Merge branch 'primary' into dev-attempt-list-react

mikedawson committed:
Update 001_010 maestro test to update path to common files

poojaustad committed:
maestro attemptlist test update

poojaustad committed:
maestro attemptlist test update

poojaustad committed:
maestro attemptlist test update

poojaustad committed:
cypress attemptlist test update

poojaustad committed:
cypress attemptlist test update

mikedawson commented:
This is obsolete - attempt list functionality is already in primray.
